### PR TITLE
[MPS] Support complex inputs to `cumprod`

### PR DESCRIPTION
**Pull request by kurtamohler**
Stack from [ghstack](https://github.com/ezyang/ghstack/tree/0.12.0) (oldest at bottom):
* __->__ #178436

Performance was measured with the following script:

<details><summary>Click to expand</summary>

```python
import torch
import torch.utils.benchmark as benchmark
import itertools

cases = [ 
    # size, dim
    [(10,), 0], 
    [(100,), 0], 
    [(1000,), 0], 
    [(10_000,), 0], 
    [(100_000,), 0], 
    [(10, 10), 0], 
    [(10, 10), 1], 
    [(100, 100), 0], 
    [(100, 100), 1], 
    [(1000, 1000), 0], 
    [(1000, 1000), 1], 
    [(10_000, 100), 1], 
    [(10_000, 100), 0], 
    [(100, 10_000), 1], 
    [(100, 10_000), 0], 
    [(10, 10, 10), 0], 
    [(10, 10, 10), 1], 
    [(10, 10, 10), 2], 
]

dtypes = [torch.float, torch.cfloat]

def run_benchmark(device):
    for dtype, (size, dim) in itertools.product(dtypes, cases):
        x = torch.randn(size, device=device, dtype=dtype)

        t = benchmark.Timer(
            stmt="torch.cumprod(x, dim=dim)",
            globals={"torch": torch, "x": x, "dim": dim},
            num_threads=torch.get_num_threads(),
        )

        for _ in range(2):
            result = t.timeit(1000)  # number of runs

        print(f'{dtype} {str(size).replace(" ", "")} {dim} {result.median:e}')

run_benchmark("mps")
```

</details>

The results are:

<details><summary>Click to expand</summary>

size | dim | float32 runtime, s | complex64 runtime, s
-- | -- | -- | --
  |   |   |  
(10,) | 0 | 9.35E-06 | 4.57E-06
(100,) | 0 | 9.40E-06 | 3.40E-06
(1000,) | 0 | 1.09E-05 | 3.44E-06
(10000,) | 0 | 1.17E-05 | 8.91E-06
(100000,) | 0 | 1.30E-05 | 5.32E-05
(10,10) | 0 | 9.57E-06 | 3.63E-06
(10,10) | 1 | 9.69E-06 | 2.83E-06
(100,100) | 0 | 1.05E-05 | 6.52E-06
(100,100) | 1 | 1.06E-05 | 3.61E-06
(1000,1000) | 0 | 9.33E-05 | 1.55E-04
(1000,1000) | 1 | 7.55E-05 | 1.41E-04
(10000,100) | 1 | 4.03E-05 | 1.39E-04
(10000,100) | 0 | 9.45E-05 | 3.78E-04
(100,10000) | 1 | 7.64E-05 | 1.44E-04
(100,10000) | 0 | 5.61E-05 | 1.49E-04
(10,10,10) | 0 | 9.67E-06 | 3.64E-06
(10,10,10) | 1 | 1.00E-05 | 3.64E-06
(10,10,10) | 2 | 9.89E-06 | 3.09E-06

</details>

### CODE REVIEW

File: aten/src/ATen/native/mps/kernels/ScanKernel.metal [hunk 1]
```diff
@@ -3,6 +3,7 @@
 using namespace metal;
 
 #include <c10/metal/common.h>
+#include <c10/metal/special_math.h>
```
Comment (malfet) [inline aten/src/ATen/native/mps/kernels/ScanKernel.metal:6]: You don't need that, do you?
```suggestion
```

**Review by malfet — APPROVED** [review]

File: aten/src/ATen/native/mps/kernels/ScanKernel.metal (hunk 1 above)
Comment (kurtamohler) [reply aten/src/ATen/native/mps/kernels/ScanKernel.metal:6]: Oh, nope. I mistakenly thought that `is_complex_v` was declared there. It doesn't build if I just remove that include, with an error saying that `is_complex_v` is undeclared. But the only reason the include fixes that error is because of a `using namespace c10::metal` statement in one of the included files. I'll just put that line in this file.
